packages/core/src/storages/request_queue_v2.ts

```diff
@@ -0,0 +1,988 @@
+        const { handledRequestCount } = await this.getInfo() ?? {};
+        return handledRequestCount ?? 0;
+    }
+
+    /**
+     * Returns an object containing general information about the request queue.
+     *
+     * The function returns the same object as the Apify API Client's
+     * [getQueue](https://docs.apify.com/api/apify-client-js/latest#ApifyClient-requestQueues)
+     * function, which in turn calls the
+     * [Get request queue](https://apify.com/docs/api/v2#/reference/request-queues/queue/get-request-queue)
+     * API endpoint.
+     *
+     * **Example:**
+     * ```
+     * {
+     *   id: "WkzbQMuFYuamGv3YF",
+     *   name: "my-queue",
+     *   userId: "wRsJZtadYvn4mBZmm",
+     *   createdAt: new Date("2015-12-12T07:34:14.202Z"),
+     *   modifiedAt: new Date("2015-12-13T08:36:13.202Z"),
+     *   accessedAt: new Date("2015-12-14T08:36:13.202Z"),
+     *   totalRequestCount: 25,
+     *   handledRequestCount: 5,
+     *   pendingRequestCount: 20,
+     * }
+     * ```
+     */
+    async getInfo(): Promise<RequestQueueInfo | undefined> {
+        return this.client.get();
+    }
+
+    /**
+     * Opens a request queue and returns a promise resolving to an instance
+     * of the {@apilink RequestQueue} class.
+     *
+     * {@apilink RequestQueue} represents a queue of URLs to crawl, which is stored either on local filesystem or in the cloud.
+     * The queue is used for deep crawling of websites, where you start with several URLs and then
+     * recursively follow links to other pages. The data structure supports both breadth-first
+     * and depth-first crawling orders.
+     *
+     * For more details and code examples, see the {@apilink RequestQueue} class.
+     *
+     * @param [queueIdOrName]
+     *   ID or name of the request queue to be opened. If `null` or `undefined`,
+     *   the function returns the default request queue associated with the crawler run.
+     * @param [options] Open Request Queue options.
+     */
+    static async open(queueIdOrName?: string | null, options: StorageManagerOptions = {}): Promise<RequestQueue> {
+        ow(queueIdOrName, ow.optional.any(ow.string, ow.null));
+        ow(options, ow.object.exactShape({
+            config: ow.optional.object.instanceOf(Configuration),
+            storageClient: ow.optional.object,
+            proxyConfiguration: ow.optional.object,
+        }));
+
+        options.config ??= Configuration.getGlobalConfig();
+        options.storageClient ??= options.config.getStorageClient();
+
+        await purgeDefaultStorages(options.config, options.storageClient);
+
+        const manager = StorageManager.getManager(this, options.config);
+        const queue = await manager.openStorage(queueIdOrName, options.storageClient);
+        queue.proxyConfiguration = options.proxyConfiguration;
+
+        return queue;
+    }
+}
+
+export { RequestQueue as RequestQueueV2 };
```

Comment:
It's exported like this due to StorageManager using `Clazz.name` for the storage client access, and before it would throw an error since storages dont have a requestQueueV2 on them

As for the duplication issues, I'll tackle those later when I'll tackle reducing this duplication out. I wanna see it working first, then clean up the diff